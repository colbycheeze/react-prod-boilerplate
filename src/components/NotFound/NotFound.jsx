import React from 'react'
import { Link } from 'react-router'
import logoSvg from 'images/logo_only_white.svg'
import classes from './notFound.scss'

const NotFound = () => (
  <div className={classes.wrapper}>
    <div className={classes.content}>

      <img className={classes.logo} src={logoSvg} alt="Medspoke Logo" />

      <div className={classes.error}>
        <div className={classes.code}>404</div>
        <div className={classes.message}>Page Not Found</div>
      </div>

      <div className={classes.options}>
        <div>Try going back or visiting the <Link to="/provider" >dashboard</Link></div>
      </div>

    </div>
  </div>
)

export default NotFound
