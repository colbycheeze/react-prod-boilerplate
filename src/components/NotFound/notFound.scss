$white: white;
$primary: blue;
$dark-blue: salmon;

.wrapper {
  display: flex;
  height: 100%;
  width: 100%;
  background-color: $primary;
  align-items: center;
  justify-content: center;
  color: white;
  text-align: center;
  background: linear-gradient(to bottom, $primary 0%,$dark-blue 100%);

  a {
    color: white;
    text-decoration: underline;
  }
}

.logo {
  margin-bottom: 3rem;
  width: 90px;
  height: 91px;
}

.error {
  border-bottom: 3px solid white;
  padding: 0 8rem 3rem;
}

.code {
  font-size: 9rem;
  font-weight: 900;
  line-height: 1.2;
}

.message {
  font-size: 4rem;
}

.options {
  padding: 3rem 3rem 15rem;
  font-size: 1.5rem;
}
