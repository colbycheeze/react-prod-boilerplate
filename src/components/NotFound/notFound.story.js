import React from 'react'
import { storiesOf } from '@kadira/storybook'
import NotFound from './NotFound'

storiesOf('404 - Not Found page', module)
  .add('404', () => (
    <NotFound />
  ))
