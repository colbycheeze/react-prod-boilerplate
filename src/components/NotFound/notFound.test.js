import React from 'react'
import NotFound from './NotFound'

describe('<NotFound />', () => {
  it('renders', () => {
    const wrapper = render(<NotFound />)

    expect(wrapper).toMatchSnapshot()
  })
})
