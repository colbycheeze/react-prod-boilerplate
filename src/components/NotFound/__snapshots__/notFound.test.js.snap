exports[`<NotFound /> renders 1`] = `
<div
  class="wrapper">
  <div
    class="content">
    <img
      alt="Medspoke Logo"
      class="logo"
      src="test-file-stub" />
    <div
      class="error">
      <div
        class="code">
        404
      </div>
      <div
        class="message">
        Page Not Found
      </div>
    </div>
    <div
      class="options">
      <div>
        Try going back or visiting the 
        <a>
          dashboard
        </a>
      </div>
    </div>
  </div>
</div>
`;
